core/citrus-base/src/main/java/com/consol/citrus/functions/core/ReadFileResourceFunction.java

```diff
@@ -49,12 +50,23 @@ public String execute(List<String> parameterList, TestContext context) {
 
         try {
             if (base64) {
-                return Base64.encodeBase64String(FileCopyUtils.copyToByteArray(FileUtils.getFileResource(parameterList.get(0), context).getInputStream()));
+                return Base64.encodeBase64String(readFileContent(parameterList, context).getBytes(FileUtils.getCharset(parameterList.get(0))));
```

Comment:
I think replacing the dynamic content by default is problematic here. This is because the file content can also be non textual e.g. an image. The dynamic replace of variables in this case will corrupt the content or fail miserably. 

This is why the base64 part of the function has skipped the replacement of variables because reading files in base64 has been used with images almost all the time.

What about introducing another optional function parameter to explicitly enable variable replacement? This could parameter could default to "true" so users can skip the replacement on demand.

Reply:
I put the default for the new parameter to false to ensure backwards compatibility.
As you noted: My test is probably the only piece of code that makes use of this feature.